src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -64,6 +65,20 @@
        (finally
          (async/unsub p# t# c#)))))
 
+(defn- restrained-drained-stream [stream max-size]
+  "Returns a stream that will throw ::body-stream-overflow and drain
+  the rest of the stream if more than max-size-data is read."
+  ;; The drain is because ruby.  i.e. if we closed the connection
+  ;; without that, one of the ruby clients wouldn't handle the broken
+  ;; pipe in a friendly way.
+  (proxy [LimitedInputStream] [stream max-size]
+    (raiseError [max-size count]
+      (loop []
+        (when-not (= -1 (.read stream))
+          (.skip stream (* 1024 1024))
```

Comment:
...perhaps not, just being cautious (could even be MAXLONG, API-wise), but that reminds me, I should double-check this with @senior  He and I dealt with some corner cases in his patch that I don't recall clearly offhand.